src/domains/global/selectors.ts

```diff
@@ -0,0 +1,45 @@
+import { prop } from "ramda"
+import { createSelector } from "reselect"
+
+import { AppStateT } from "store/app-state"
+
+import { GetKeyArguments, getKeyForCommonColorsState } from "./reducer"
+import { storeKey } from "./constants"
+
+export const createSelectAssignedColors = (args: GetKeyArguments) => (state: AppStateT) => {
+  const keyName = getKeyForCommonColorsState(args)
+  const substate = state[storeKey].commonColorsKeys[keyName]
+  return substate && substate.assigned
```

Comment:
@Azein re: your comment from previous CR:
> For variable-dependant selection and &&-based protection just the R.path could be used, I think.

I have tried R.path but the typing doesn't work that well in that scenario, how about if we use optional chaining here? (that feature is not yet added to the codebase)

Reply:
I'm all for it, we are adopting it in cloud-fe currently, works nice.